test/unit/host_discovered_test.rb

```diff
@@ -24,12 +23,43 @@ class HostDiscoveredTest < ActiveSupport::TestCase
     assert Host::Discovered.find_by_name('mace41f13cc3658')
   end
 
+  test "should setup subnet" do
+    raw = parse_json_fixture('/facts.json')
+    subnet = FactoryGirl.create(:subnet)
+    Subnet.expects(:subnet_for).with('10.35.27.3').returns(subnet)
+    subnet.stubs(:organizations).returns([Organization.first])
+    subnet.stubs(:locations).returns([Location.first])
```

Comment:
Why stub the org/loc? This could do, I think: 

```ruby
subnet = FactoryGirl.create(:subnet, :organizations => [Organization.first], :locations => [Location.first])
```